pylint/checkers/classes/class_checker.py

```diff
@@ -880,6 +887,50 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
         self._check_proper_bases(node)
         self._check_typing_final(node)
         self._check_consistent_mro(node)
+        self._check_declare_non_slot(node)
+
+    def _check_declare_non_slot(self, node: nodes.ClassDef) -> None:
+        if not self._has_valid_slots(node):
+            return
+
+        slot_names = self._get_classdef_slots_names(node)
+
+        # Stop if empty __slots__ in the class body, this likely indicates that
+        # this class takes part in multiple inheritance with other slotted classes.
+        if not slot_names:
+            return
+
+        # Stop if we find __dict__, since this means attributes can be set
+        # dynamically
+        if "__dict__" in slot_names:
+            return
+
+        for base in node.bases:
+            ancestor = safe_infer(base)
+            if not isinstance(ancestor, nodes.ClassDef):
+                continue
+            # if any base doesn't have __slots__, attributes can be set dynamically, so stop
+            if not self._has_valid_slots(ancestor):
+                return
+            for slot_name in self._get_classdef_slots_names(ancestor):
+                if slot_name == "__dict__":
+                    return
+                slot_names.append(slot_name)
+
+        # Every class in bases has __slots__, our __slots__ is non-empty and there is no __dict__
+
+        for child in node.body:
+            if isinstance(child, nodes.AnnAssign):
+                if child.value is not None:
+                    continue
```

Comment:
We have ``self.a = 42`` line 183 in the functional test, maybe getting coverage for this is as simple as adding a line with ``self.b : str = "AnnAssign.value is not None"`` too ?

Reply:
Thanks, I've added this, hopefully this will fix the code coverage complaint. Waiting for the CI checks to run now.

Reply:
There might be specific condition to enter this condition, (a "b" value in slot maybe ?)

Reply:
The codecov test result seems to have changed now, it's claiming that a line isn't covered which must be covered by the tests which generate the error. I don't understand why the [codecov test](https://github.com/pylint-dev/pylint/pull/9564/checks?check_run_id=25859465941) is failing, are you able to shed any light on this please?

Reply:
I tried something with ``b: str`` being declared. I assume the diff between main and this branch got too big. Also fixed the doc generation.